front/pages/api/w/[wId]/builder/agents/[aId]/actions.ts

```diff
@@ -0,0 +1,87 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import type { AgentBuilderAction } from "@app/components/agent_builder/AgentBuilderFormContext";
+import { transformActionsToFormData } from "@app/components/agent_builder/transformAgentConfiguration";
+import { getAgentConfiguration } from "@app/lib/api/assistant/configuration";
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/auth_wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { apiError } from "@app/logger/withlogging";
+import type { WithAPIErrorResponse } from "@app/types";
+
+export type GetAgentActionsResponseBody = {
+  actions: AgentBuilderAction[];
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<GetAgentActionsResponseBody>>,
+  auth: Authenticator
+): Promise<void> {
+  if (req.method !== "GET") {
+    return apiError(req, res, {
+      status_code: 405,
+      api_error: {
+        type: "method_not_supported_error",
+        message: "The method passed is not supported, GET is expected.",
+      },
+    });
+  }
+
+  const { aId } = req.query;
+  if (typeof aId !== "string") {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid agent ID.",
+      },
+    });
+  }
+
+  try {
+    const agentConfiguration = await getAgentConfiguration(auth, aId, "full");
+    if (!agentConfiguration) {
+      return apiError(req, res, {
+        status_code: 404,
+        api_error: {
+          type: "agent_configuration_not_found",
+          message: "Agent configuration not found.",
+        },
+      });
+    }
+
+    const actionsResult = transformActionsToFormData(
```

Comment:
It feels a bit weird to me, do we want to transform this on server side and not client? 🤔

Reply:
Also we currently fetch dataSourceViews, mcpServerViews  etc for assistant builder action fetch, does this mean we don't have to do it there?

Reply:
Also x 2, this doesn't seem to be working? I don't see any actions when opening an existing agent?